cli/cli_doc_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli_test
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+type EatCommand struct {
+	cli.BaseCommand
+}
+
+func (c *EatCommand) Desc() string {
+	return "Eat some food"
+}
+
+func (c *EatCommand) Help() string {
+	return strings.Trim(`
+The eat command eats food.
+`+c.Flags().Help(), "\n")
+}
+
+func (c *EatCommand) Flags() *cli.FlagSet {
+	return cli.NewFlagSet()
+}
+
+func (c *EatCommand) Run(ctx context.Context, args []string) error {
+	// TODO: implement
```

Comment:
nit: both examples could be combined into a simple example that uses a flag and has a valid run function, even something trivial like calculating the fibonacci number of N where N is an argument?